From 11ce11a0a2e13a224813ee9be784d246bab2a522 Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Fri, 20 Jun 2014 20:32:18 -0700
Subject: [PATCH] configure.ac: use pkg-config to check xml2

The xml2-config has gone, see oe-core's 5870bd272b0b077d0826fb900b251884c1c05061

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 configure.ac |   14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/configure.ac b/configure.ac
index 95d7276..3a13f4e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -491,7 +491,6 @@ AC_PATH_PROGS(IFCONFIG, ifconfig, /sbin/ifconfig)
 AC_PATH_PROGS(MAILCMD, mailx mail)
 AC_PATH_PROGS(EGREP, egrep)
 AC_PATH_PROGS(PKGCONFIG, pkg-config)
-AC_PATH_PROGS(XML2CONFIG, xml2-config)
 
 AC_ARG_ENABLE([doc],
  AS_HELP_STRING([--enable-doc], [build documentation (default is yes)]),
@@ -694,15 +693,12 @@ dnl AC_CHECK_HEADERS(netinet/ip6.h)
 dnl AC_CHECK_HEADERS(netinet/ip_icmp.h)
 
 AC_MSG_CHECKING(for special libxml2 includes)
-if test "x$XML2CONFIG" = "x"; then
-   AC_MSG_ERROR(libxml2 config not found)
-else
-   XML2HEAD="`$XML2CONFIG --cflags`"
-   AC_MSG_RESULT($XML2HEAD)
-   AC_CHECK_LIB(xml2, xmlReadMemory)
+PKG_CHECK_MODULES(PKG_XML2, [libxml-2.0],,)
+if test "x$PKG_XML2_CFLAGS" != "x"; then
+  CPPFLAGS="$CPPFLAGS $PKG_XML2_CFLAGS"
+  AC_MSG_RESULT($PKG_XML2_CFLAGS)
 fi
-
-CPPFLAGS="$CPPFLAGS $XML2HEAD"
+AC_CHECK_LIB(xml2, xmlReadMemory)
 
 AC_CHECK_HEADERS(libxml/xpath.h)
 if test "$ac_cv_header_libxml_xpath_h" != "yes"; then
-- 
1.7.9.5

